### PR TITLE
DOC: Mention PyArray_DIMS can be NULL

### PR DESCRIPTION
**Pull request by kritisingh1**
Related to #13654

### CODE REVIEW

**Comment by eric-wieser** [comment]
It can only be null on 0d arrays

**Comment by kritisingh1** [comment]
Wouldn't it be `0` for 0d arrays?

**Comment by eric-wieser** [comment]
You're thinking of `PyArray_NDIM` which counts the dimensions, `DIMS` is a pointer to the array of dimensions.

File: doc/source/reference/c-api.array.rst [hunk 1]
```diff
@@ -33,7 +33,7 @@ sub-types).
 
     Returns a pointer to the dimensions/shape of the array. The
     number of elements matches the number of dimensions
-    of the array.
+    of the array. Returns ``NULL`` for 0-dimensional arrays.
```
Comment (mattip) [inline doc/source/reference/c-api.array.rst]: Should be qualified: "Can return ``NULL`` ...

**Pull request merged**